pkg/controller/pxc/volumes.go

```diff
@@ -244,10 +244,15 @@ func (r *ReconcilePerconaXtraDBCluster) reconcilePersistentVolumes(ctx context.C
 		return nil
 	}
 
-	err = k8s.AnnotateObject(ctx, r.client, cr, map[string]string{pxcv1.AnnotationPVCResizeInProgress: metav1.Now().Format(time.RFC3339)})
+	now := metav1.Now().Format(time.RFC3339)
+
+	err = k8s.AnnotateObject(ctx, r.client, cr, map[string]string{pxcv1.AnnotationPVCResizeInProgress: now})
 	if err != nil {
 		return errors.Wrap(err, "annotate pxc")
 	}
+	// AnnotateObject will not add an annotation to the current cr reference.
+	// We should annotate it manually so that other reconcile functions can see it.
+	cr.Annotations[pxcv1.AnnotationPVCResizeInProgress] = now
```

Comment:
maybe `AnnotateObject` should do this? this behavior might be unexpected and might confuse other people too

Reply:
https://github.com/percona/percona-xtradb-cluster-operator/pull/2069/commits/3efdcb2cbe023707586ad04e06faefadafc95c22